### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Limits.Shapes.Kernels

### PR DESCRIPTION
**Pull request by mattrobball**
---

- [x] depends on: #2622
[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2622
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2622~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/CategoryTheory/Limits/Shapes/Kernels.lean [hunk 1]
```diff
@@ -0,0 +1,1143 @@
...
+  HasLimit.isoOfNatIso (by rw [h])
+#align category_theory.limits.kernel_iso_of_eq CategoryTheory.Limits.kernelIsoOfEq
+
+@[simp]
+theorem kernelIsoOfEq_refl {h : f = f} : kernelIsoOfEq h = Iso.refl (kernel f) := by
+  ext
+  simp [kernelIsoOfEq]
+#align category_theory.limits.kernel_iso_of_eq_refl CategoryTheory.Limits.kernelIsoOfEq_refl
+
+/- Porting note: indunction on Eq is trying instantiate another g...-/
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Shapes/Kernels.lean:333]:
```suggestion
/- Porting note: induction on Eq is trying instantiate another g...-/
```

**Review by jcommelin — APPROVED** [review]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Pull request closed without merging**